test/sql/insert_sql_test.cpp

```diff
@@ -570,5 +570,42 @@ TEST_F(InsertSQLTests, UniqueColumn) {
   txn_manager.CommitTransaction(txn);
 }
 
+TEST_F(InsertSQLTests, BadTypes) {
+  auto &txn_manager = concurrency::TransactionManagerFactory::GetInstance();
+  auto txn = txn_manager.BeginTransaction();
+  catalog::Catalog::GetInstance()->CreateDatabase(DEFAULT_DB_NAME, txn);
+  txn_manager.CommitTransaction(txn);
+  std::string error_message;
+  std::unique_ptr<optimizer::AbstractOptimizer> optimizer(
+      new optimizer::Optimizer());
+
+  std::string create_table("CREATE TABLE foo (id1 int, id2 bigint,"
+                                         "id3 smallint, id4 tinyint,"
+                                         "id5 decimal);");
+  TestingSQLUtil::ExecuteSQLQuery(create_table);
+  // Insert an unconvertible int.
+  std::string query("INSERT INTO(id) foo VALUES('h');");
```

Comment:
What is this insert trying to test? Is this what you intended?

Reply:
It's trying to test whether an exception is raised when an invalid value is inserted.